### PR TITLE
Add optional meta to DocBase

### LINKED ISSUE #36: Shouldn't DocWithData have `meta?: MetaObject` included?
**Is your feature request related to a problem? Please describe.**

I regularly need to process jsonapi responses that may or may not include the top level `meta` key. However, `DocWithData` does not include the optional `meta` key.

**Describe the solution you'd like**

Add `meta?: MetaObject` to `DocWithData`. Or better yet, to `DocBase`.

I may be missing something if you intentionally did not include it in that way, and decided to provide `DocWithMeta` as a separate entity, but the more I read the specification, `meta` can be found in any valid jsonapi response, or at least in responses with `data`, as seen [here](https://jsonapi.org/format/#document-meta).

**Describe alternatives you've considered**

I was thinking of declaring a type `DocWithData & DocWithMeta`, but that does not express that `meta` is optional. I'm left with declaring something like this:

```typescript
type MyDocWithData = DocWithData & { meta?: MetaObject };
```

But then again I'd have to replicate the above for `SingleResourceDoc` and `CollectionResourceDoc`.

If this is accepted, I'm more than happy to work on a PR myself.

### PR DESCRIPTION
**Pull request by gnapse**
Closes #36 

See discussion there for details.

### CODE REVIEW

**Comment by mike-north** [comment]
Please reword your commit message to 

```
fix: add optional meta to DocBase
```
per https://conventionalcommits.org

**Comment by gnapse** [comment]
Done!

I'd recommend that this should be explicitly stated somewhere. Ideally a `CONTRIBUTING.md` file in the root of the project (which BTW GitHub notices and highlights its existence to users that are making their first contribution to a repo).

**Comment by mike-north** [comment]
> I'm more than happy to make a PR for this as well.
Yes please!

File: index.ts [hunk 1]
```diff
@@ -33,6 +33,7 @@ export interface DocWithErrors extends DocBase {
 export interface DocBase {
 	jsonapi?: ImplementationInfo;
 	links?: Links | PaginationLinks;
+	meta?: MetaObject;
```
Comment (mike-north) [inline index.ts]: Can you write some tests for this, following examples like this
https://github.com/mike-north/jsonapi-typescript/blob/master/tests/resources/main.test.ts

Comment (gnapse) [reply index.ts]: Will do.

Nice, I'm even learning how to test TS! 😃

**Comment by mike-north** [comment]
All this should need is some tests :)

**Comment by gnapse** [comment]
Yes, sorry, I haven't been able to devote time to it. Thanks for the reminder. Will try to tackle it before next week.

**Comment by johannesschobel** [comment]
just pushing this upwards @gnapse :laughing:

**Comment by johannesschobel** [comment]
This can be closed because of #62

**Pull request closed without merging**